control-plane/agents/src/bin/core/controller/reconciler/snapshot/garbage_collector.rs

```diff
@@ -181,3 +182,30 @@ async fn prune_volume_snapshot_reconciler(
 
     Ok(PollerState::Idle)
 }
+
+#[tracing::instrument(skip(snapshot, context), level = "trace", fields(snapshot.id = %snapshot.uuid(), request.reconcile = true))]
+async fn delete_no_transaction_volume_snapshot_reconciler(
+    snapshot: &mut OperationGuardArc<VolumeSnapshot>,
+    context: &PollContext,
+) -> PollResult {
+    // Only proceed if the spec is in creating and has no transactions.
+    if !snapshot.as_ref().status().creating()
+        && !snapshot.as_ref().metadata().transactions().is_empty()
+    {
+        return Ok(PollerState::Idle);
+    }
+
+    let Some(snap_rsc) = context.specs().volume_snapshot_rsc(snapshot.uuid()) else {
+        return Ok(PollerState::Idle);
+    };
+
+    snapshot
```

Comment:
Should we try to get the volume here and delete through the volume if available?

Reply:
Good point.